### DISCUSSION TITLE
'Show Call Hierarchy' should combine results from multiple language services

### DISCUSSION THREAD

**Discussion by nayeemrmn** [microsoft/vscode-discussions#868] [Extension Development QnA] [answered]
### Context

The Deno extension document-selects TypeScript files to send to the Deno LSP. We interoperate it with the built-in `typescript-language-features` extension, and for most purposes this works.

For Deno-relevant scripts we handle requests in the LSP and use a plugin for `typescript-language-features` to make its responses empty. For Node-relevant scripts we send empty responses from the Deno LSP and allow the `typescript-language-features` responses through.

---

However the 'Show Call hierarchy' feature will only ask the Deno LSP for `callHierarchy/incomingCalls` if that language client is registered. It results in this bug for Node users who have the Deno extension enabled, and is irresolvable if you're trying to interoperate the two in a monorepo: 
- https://github.com/denoland/vscode_deno/issues/765.

The 'Find All References' feature on the other hand works great, because it combines results from all relevant language servers. So it would be nice if 'Show Call Hierarchy' worked the same. The logic should just be a matter of concatenating arrays.

I want to see if this would be welcome or if there are alternative solutions before feature-requesting it in the main repo.

**Comment by olguzzar** [comment, marked as answer]
Hi @nayeemrmn, thanks for posting! Seems like the best course of action would be to go ahead and file the feature request on that repo, as that will get more relevant eyes on the topic and can begin the discussion there. They also have [discussions enabled on the repo](https://github.com/denoland/vscode_deno/discussions), though it doesn't seem particularly active.
